frontend/src/views/EnvironmentDetail.vue

```diff
@@ -169,17 +169,46 @@ const doUpdate = (environmentPatch: Environment) => {
     pendingUpdate.tier = environmentPatch.tier;
   }
 
-  environmentV1Store.updateEnvironment(pendingUpdate).then((environment) => {
-    assignEnvironment(environment);
+  environmentV1Store
+    .updateEnvironment(pendingUpdate)
+    .then((environment) => {
+      assignEnvironment(environment);
 
-    pushNotification({
-      module: "bytebase",
-      style: "SUCCESS",
-      title: t("environment.successfully-updated-environment", {
-        name: environment.title,
-      }),
+      const disallowed = environment.tier === EnvironmentTier.PROTECTED;
+      if (disallowed) {
+        return policyV1Store.upsertPolicy({
+          parentPath: environment.name,
+          updateMask: ["payload", "inherit_from_parent"],
+          policy: {
+            type: PolicyTypeV1.ACCESS_CONTROL,
+            inheritFromParent: true,
+            accessControlPolicy: {
+              disallowRules: [{ fullDatabase: true }],
+            },
+          },
+        });
+      } else {
+        policyV1Store
+          .getOrFetchPolicyByParentAndType({
+            parentPath: state.environment.name,
```

Comment:
using `environment.name` might be more consistent

Reply:
Agreed